### PR TITLE
Add internal build information (build classpath, implicit imports) to MinimalBuildStructure

### PR DESCRIPTION
**Pull request by dancingrobot84**

### CODE REVIEW

**Comment by jsuereth** [comment]
This LGTM, is there anything else we may need in here?  (cc @eed3si9n )

File: commons/protocol/src/main/scala/sbt/protocol/Keys.scala [hunk 1]
```diff
@@ -111,8 +112,17 @@ object MinimalProjectStructure {
   implicit val pickler: Pickler[MinimalProjectStructure] = genPickler[MinimalProjectStructure]
 }
 
+final case class BuildData(
+  uri: URI,
+  classpath: Vector[File],
+  imports: Vector[String])
+object BuildData {
+  implicit val picklerUnpickler = PicklerUnpickler.generate[BuildData]
```
Comment (eed3si9n) [inline commons/protocol/src/main/scala/sbt/protocol/Keys.scala]: Implicit val should have type annotation.

File: protocol-test/src/test/resource/saved-protocol/3/complex/build_structure.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"builds":["file:\/\/\/test\/project"],"projects":[{"id":{"build":"file:\/\/\/test\/project","name":"test"},"plugins":["com.foo.Plugin"],"dependencies":null,"$type":"sbt.protocol.MinimalProjectStructure"}],"$type":"sbt.protocol.MinimalBuildStructure"}
```
Comment (eed3si9n) [inline protocol-test/src/test/resource/saved-protocol/3/complex/build_structure.json]: Where's buildsData here?

**Comment by eed3si9n** [comment]
I pulled the PR down to my local machine to play around.

```scala
scala> import sbt.serialization._
import sbt.serialization._

scala> import sbt.protocol.MinimalBuildStructure
import sbt.protocol.MinimalBuildStructure

scala> MinimalBuildStructure(Vector(), Vector(), Vector())
res0: sbt.protocol.MinimalBuildStructure = MinimalBuildStructure(Vector(),Vector(),Vector())

scala> toJsonString(res0)
res1: String = {"builds":[],"buildsData":[],"projects":[],"$type":"sbt.protocol.MinimalBuildStructure"}

scala> val protocol2Data = """{"builds":["file:\/\/\/test\/project"],"projects":[{"id":{"build":"file:\/\/\/test\/project","name":"test"},"plugins":["com.foo.Plugin"],"dependencies":null,"$type":"sbt.protocol.MinimalProjectStructure"}],"$type":"sbt.protocol.MinimalBuildStructure"}"""
protocol2Data: String = {"builds":["file:\/\/\/test\/project"],"projects":[{"id":{"build":"file:\/\/\/test\/project","name":"test"},"plugins":["com.foo.Plugin"],"dependencies":null,"$type":"sbt.protocol.MinimalProjectStructure"}],"$type":"sbt.protocol.MinimalBuildStructure"}

scala> fromJsonString[MinimalBuildStructure](protocol2Data)
res2: scala.util.Try[sbt.protocol.MinimalBuildStructure] = Success(MinimalBuildStructure(Vector(file:///test/project),Vector(),Vector(MinimalProjectStructure(ProjectReference(file:///test/project,test),Vector(com.foo.Plugin),None))))
````

Generally I'm happy to see that protocol2 data is deserializable using the new case class.
I'm concerned that the newly added buildsData is missing from one-way test data, as commented.

**Comment by dancingrobot84** [comment]
@eed3si9n I've fixed things you've mentioned and updated protocol v3 tests for build structure change event as well.

**Comment by eed3si9n** [comment]
@dancingrobot84 Thanks. LGTM

**Pull request merged**